docs/source/conf.py

```diff
@@ -308,3 +311,29 @@
 
 # How to display URL addresses: 'footnote', 'no', or 'inline'.
 #texinfo_show_urls = 'footnote'
+
+
+commit_hash = os.environ.get("SOPEL_GIT_COMMIT", "")[:7]
+is_dirty = bool(os.environ.get("SOPEL_GIT_DIRTY"))
+if commit_hash:
+    github_ref = "https://github.com/sopel-irc/sopel/commit/{commit_hash}".format(commit_hash=commit_hash)
```

Comment:
It occurred to me while preparing my revisions that this GitHub link isn't guaranteed to be valid when building the preview docs against commits that aren't pushed to the repo, but I'm not sure that's worth worrying about, it should be valid in general for work we do in this repo (and the docs that end up on the website), and anybody hacking on a local clone of the source probably will understand if the link is invalid.